src/porepy/numerics/time_step_control.py

```diff
@@ -435,16 +450,31 @@ def next_time_step(
 
         return self.dt
 
-    def _adaptation_based_on_iterations(self, iterations: int) -> None:
+    def increase_time(self) -> None:
```

Comment:
@jhabriel why are increase_time and increase_time_index public?
Shouldn't only the tsc object call them and handle time increase during the request for the next time step ?

Reply:
Good question. This was also the way I implanted first, i.e. to advance in time after computing the next time step as part of `next_time_step()`

However, I decided against that approach for two reasons: (1) in the time loop, the simulation time must be increased before solving the model and only then the next time step should be computed, and (2) marching in time should be independent of the way the time step is computed. 

Decoupling these steps could be also useful in situations where we are only interested in knowing what would the next time step be without taking any other action.

Reply:
Okay I understand.

My motivation to combine that comes from the fact that the tsc class has all the information it needs to be able to decide if time should be advanced or not. So the usage of tsc would be simplified to almost one method call.

But I understand that you want to be able to get an outlook on what the next dt would be.